### PR TITLE
Makozlo/upgrade react native

### LINKED ISSUE #8: Update React Native to latest version
Need to update to the newest stable release available so new projects start off up-to-date
https://github.com/facebook/react-native/releases/tag/v0.64.2

### PR DESCRIPTION
**Pull request by makozlo**
This PR: 
- upgrades the `react-native` version in both templatesto 0.64.2
- removes references to the `propeller` template
- adds the `react-native-clean-project` package to `devDependencies`.
  - `react-native-clean-project` will replace the cleaning scripts that we have.

### CODE REVIEW

File: .prettierrc.js [hunk 1]
```diff
@@ -3,4 +3,5 @@ module.exports = {
   singleQuote: true,
   printWidth: 100,
   trailingComma: 'all',
+   arrowParens: 'avoid',
```
Comment (townofdon) [inline .prettierrc.js]: My personal pref is to have arrowParens as `always`. Reason being - less refactoring, and less diffs created if a function signature changes. What's the rationale for using `avoid` here?

File: templates/jet/.prettierrc.js [hunk 2]
```diff
@@ -3,4 +3,5 @@ module.exports = {
   singleQuote: true,
   printWidth: 100,
   trailingComma: 'all',
+  arrowParens: 'avoid',
```
Comment (townofdon) [inline templates/jet/.prettierrc.js]: Same question - could we change this to `always`?

**Review by townofdon — CHANGES_REQUESTED** [review]
I'm cool with everything except for the `arrowParens` choice. Unless there's a good reason for using `avoid` I'm of the opinion that we should go with `always` for better code consistency. Helps with clearer Git diffs and also with quicker refactors.

File: .prettierrc.js (hunk 1 above)
Comment (makozlo) [reply .prettierrc.js]: Thanks for mentioning this - this was part of the React Native upgrade for some reason. I'm with you though, I'd rather have parentheses always.

File: templates/jet/.prettierrc.js (hunk 2 above)
Comment (makozlo) [reply templates/jet/.prettierrc.js]: Same as above - this was part of the React Native upgrade for some reason. I'm with you though, I'd rather have parentheses always.

**Review by townofdon — APPROVED** [review]
Looks good! Have not tested however.

**Pull request merged**